### ISSUE TITLE
clang: using enum tdef rejected

### ISSUE THREAD

**Issue by urnathan** [llvm/llvm-project#57659]
DR2621  notes implementation divergence for:
```cpp
enum class E {
  a, b, c
};

using MyE = E;

int main() {
  using enum MyE;   // #1
}
```
`clang++ -std=c++20 -c x.cc`
```console
x.cc:8:14: error: type alias 'MyE' cannot be referenced with a enum specifier
  using enum MyE;   // #1
             ^
x.cc:5:7: note: declared here
using MyE = E;
      ^
```
(gcc, EDG, MSVC accept).  It should be accepted.

**Comment by urnathan** [comment]
```
enum class E {  a };
int E; // #2
void f () {
  using enum E;  // ill-formed, finds #2
  using enum enum E; // ICE
}
```

**Comment by urnathan** [comment]
fixed 3d2080683f1d 2022-09-14 | [clang][DR2621] using enum NAME lookup fix
